### DISCUSSION TITLE
Big Sur update errors with gem

### DISCUSSION THREAD

**Discussion by ppsirius** [Homebrew/discussions#927] [Everyday usage]
After update to Big Sur 11.2.1, I get errors on my brew 😬

`brew doctor                              
Ignoring eventmachine-1.2.7 because its extensions are not built. Try: gem pristine eventmachine --version 1.2.7
Ignoring ffi-1.13.1 because its extensions are not built. Try: gem pristine ffi --version 1.13.1
Ignoring http_parser.rb-0.6.0 because its extensions are not built. Try: gem pristine http_parser.rb --version 0.6.0
Ignoring sassc-2.4.0 because its extensions are not built. Try: gem pristine sassc --version 2.4.0
==> Installing 'bundler' gem
Ignoring eventmachine-1.2.7 because its extensions are not built. Try: gem pristine eventmachine --version 1.2.7
Ignoring ffi-1.13.1 because its extensions are not built. Try: gem pristine ffi --version 1.13.1
Ignoring http_parser.rb-0.6.0 because its extensions are not built. Try: gem pristine http_parser.rb --version 0.6.0
Ignoring sassc-2.4.0 because its extensions are not built. Try: gem pristine sassc --version 2.4.0
`
I tried with  sudo xcode-select --install but without any success. On brew upgrade got the same errors.

**Comment by carlocab** [comment]
Try uninstalling the gems that are mentioned as being ignored because their extensions are not built.

**Reply by ppsirius** [reply]
I uninstalled the packages but now have something like this:
`brew upgrade
==> Installing 'bundler' gem
Error: failed to install the 'bundler' gem.`

**Reply by carlocab** [reply]
Try
```
rm -rf ~/.gem
git -C $(brew --repo) clean -xdf
brew update-reset
```

**Reply by octoer** [reply]
> Try
> 
> ```
> rm -rf ~/.gem
> git -C $(brew --repo) clean -xdf
> brew update-reset
> ```

I tried it, but it didn't work.

**Reply by zxcstc1256** [reply]
I got the error like #923. I fixed it. 
At first, I check `CommandLineTools` path:

> xcrun --sdk macosx --show-sdk-path
xcrun: error: active developer path ("/Applications/Xcode.app/Contents/Developer") does not exist
Use `sudo xcode-select --switch path/to/Xcode.app` to specify the Xcode that you wish to use for command line developer tools, or use `xcode-select --install` to install the standalone command line developer tools.
See `man xcode-select` for more details.

Because I uninstalled `Xcode.app`, and just installed `CommandLineTools` with `xcode-select  --install`.

> sudo xcode-select -switch /

It did work.

**Reply by ppsirius** [reply]
> Try
> 
> ```
> rm -rf ~/.gem
> git -C $(brew --repo) clean -xdf
> brew update-reset
> ```

It's not working for me. When I check path of xcode I get:
`/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk`
but the version of my system is Big Sur 11.2.1, it is not a problem?

**Reply by carlocab** [reply]
What does
```
ls -l /Library/Developer/CommandLineTools/SDKs
```
say?

**Reply by octoer** [reply]
Guys, when I tried reinstalling CommandLineTools, the problem was solved

**Reply by ppsirius** [reply]
> ls -l /Library/Developer/CommandLineTools/SDKs

` ls -l /Library/Developer/CommandLineTools/SDKs
total 0
lrwxr-xr-x  1 root  wheel   14 Feb 24 20:30 MacOSX.sdk -> MacOSX11.1.sdk
drwxr-xr-x  8 root  wheel  256 Jul  9  2020 MacOSX10.15.sdk
drwxr-xr-x  7 root  wheel  224 Nov 30 13:33 MacOSX11.1.sdk`

**Reply by carlocab** [reply]
Ok, still not sure what the problem is. Try reinstalling your CLT:
```
sudo rm -rf /Library/Developer/CommandLineTools
xcode-select --install
```

**Reply by ppsirius** [reply]
> Ok, still not sure what the problem is. Try reinstalling your CLT:
> 
> ```
> sudo rm -rf /Library/Developer/CommandLineTools
> xcode-select --install
> ```

I reinstall it with success but still have the same behavior:
`brew upgrade
Updating Homebrew...
==> Installing 'bundler' gem
Error: failed to install the 'bundler' gem.
==> Installing 'bundler' gem
Error: failed to install the 'bundler' gem.`

**Reply by Bo98** [reply]
Try `brew update` now. We've made changes which mean `bundler` shouldn't be required anymore for most operations.

**Reply by ppsirius** [reply]
> Try `brew update` now. We've made changes which mean `bundler` shouldn't be required anymore for most operations.

Still have the same
`brew update
==> Installing 'bundler' gem
Error: failed to install the 'bundler' gem.`

**Reply by Bo98** [reply]
Might need to `brew update-reset`.

**Reply by YJPL** [reply]
> Might need to `brew update-reset`.

This worked for me, thank you so much.

**Comment by tylercheung** [comment]
I sort of ran into something like this when I accidentally tried doing gem update --system or something like that, and messed up all the default OS X gems...

**Comment by BerginJay** [comment]
Simply using `gem pristine X --version x.x.x` is not working may because your system gem is not your Homebrew gem...

First, run `brew config` and check your **Homebrew Ruby** path. It may like this:
```Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby```.

Then replace ruby with gem, run pristine command:
```/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/gem pristine X --version x.x.x```.

This is how I solve my problem.

**Comment by milochen0418** [comment]
This is work for me 
$ sudo gem install cocoapods

**Reply by abhinand-kv** [reply]
sudo gem install cocoapods
This solved my issue
